docs/logging.md

```diff
@@ -0,0 +1,204 @@
+# Contributor Guidelines - Logging
+
+## Framework
+
+We use structured logging, and specifically we use [Zap](https://github.com/uber-go/zap)
+as our logging framework.
+
+Why not something else?
+
+- [Zerolog](https://github.com/rs/zerolog) is more performant and better
+  looking, but its syntax is more error-prone (no logging in case you forget
+  to call `Msg("...")` at the end)
+- Go's [Slog](https://pkg.go.dev/log/slog) is new, slower (according to Zap
+  benchmarks), and doesn't provide useful benefits
+
+## Default Level
+
+The default minimum logging level is [Info](https://pkg.go.dev/go.uber.org/zap/zapcore#InfoLevel).
+To change this, set the following AutoKitteh config value to the lower-case or
+all-caps name of the level you want. For example:
+
+- CLI: `ak config set logger.level debug`
+
+- Config YAML file:
+
+  ```yaml
+  logger:
+    level: debug
+  ```
+
+- Environment variable: `AK_LOGGER__LEVEL=DEBUG`
+
+## Logger Naming
+
+- Local variables: `l` is preferable when its meaning is clear, otherwise
+  `logger` is acceptable too
+
+  - `z` is discouraged - we use Zap, but we don’t want the code to be tied to
```

Comment:
This will require a lot of code changes. I counted 63 places in the code that use `z`.

Reply:
I think having `z` is a good marker fo what we haven't overviewed yet :) WIll use `l` from now on.